### PR TITLE
Add python solution using Dijkstra's algorithm

### PR DESCRIPTION
**Pull request by HerbertMcSnout**
This does not pass the automatic check script, so it is possible that there is a bug, but it does seem to produce the correct results.

### CODE REVIEW

File: challenge04/solution-herbertmcsnout.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+from scipy.sparse import csr_matrix
```
Comment (pbui) [inline challenge04/solution-herbertmcsnout.py:3]: Ah, dredd doesn't have scipy.

Comment (HerbertMcSnout) [reply challenge04/solution-herbertmcsnout.py:3]: Ohhh, I'll just implement Dijkstra's alg by hand then

**Comment by pbui** [comment]
I have verified this offline, so I will merge it!

File: challenge04/solution-herbertmcsnout.py (hunk 1 above)
Comment (pbui) [reply challenge04/solution-herbertmcsnout.py:3]: Up to you, you can make another PR with the change :]

**Pull request merged**